.cursor/rules/api-specific-config.mdc

```diff
@@ -57,6 +72,32 @@ alwaysApply: false
   - Use `| null` not `| undefined` for nullable database fields
   - Consider using Result types for operations that may fail
 
+## Atomic Operations & Race Condition Prevention
+
+### Service Layer Atomicity
+- **TOCTOU Prevention**: Avoid Time-Of-Check-Time-Of-Use patterns
+  - ❌ Bad: `const comp = await getCompetition(); if (comp.type === 'perps') {...}`
+  - ✅ Good: `const isPerps = await isCompetitionType(id, 'perps'); if (isPerps) {...}`
+- **Atomic Check Methods**: Create single-purpose methods that combine fetch and validation
```

Comment:
yeah I still kind of disagree with this one.  This doesn't make anything safer since the value in the db can still be changed right after you check it, unless you're using a row lock, but I don't actually think we want to be using row locks everywhere and have to deal with timeouts, retries and everything that comes with that, not to mention the extra load on the db.

I think the real fixes for these types of issues are going to come on the side of the database *writes*, not the reads, where the writes should also be verifying their pre-conditions are still true and haven't changed out from under them since the code originally checked them